zokrates_core/src/flatten/mod.rs

```diff
@@ -402,54 +403,29 @@ impl Flattener {
                 // Handle complex parameters and assign values:
                 // Rename Parameters, assign them to values in call. Resolve complex expressions with definitions
                 for (i, param_expr) in param_expressions.clone().into_iter().enumerate() {
-                    let new_var;
-                    let param_expr = param_expr.apply_substitution(&self.substitution);
-
-                    match param_expr {
+                    let rhs = match param_expr {
                         TypedExpression::FieldElement(e) => {
-                            match e {
-                                FieldElementExpression::Identifier(id) => {
-                                    new_var = format!("{}param_{}", &prefix, id);
-                                    statements_flattened
-                                        .push(FlatStatement::Definition(new_var.clone(), FlatExpression::Identifier(id.clone().to_string())));
-                                },
-                                _ => {
-                                    // for field elements, flatten the input and assign it to a new variable
-                                    let expr_subbed = e.apply_substitution(&self.substitution);
-                                    let rhs = self.flatten_field_expression(
-                                        functions_flattened,
-                                        arguments_flattened,
-                                        statements_flattened,
-                                        expr_subbed,
-                                    );
-                                    new_var = format!("{}param_{}", &prefix, i);
-                                    statements_flattened
-                                        .push(FlatStatement::Definition(new_var.clone(), rhs));
-                                }
-                            }
+                            // for field elements, flatten the input and assign it to a new variable
```

Comment:
@JacobEberhardt just wanna make sure I'm not missing something here, any thoughts on why matching Identifier separately above was necessary?

Reply:
Not necessary, new version is fine.